src/herder/Upgrades.cpp

```diff
@@ -163,50 +249,60 @@ Upgrades::getParameters() const
 }
 
 std::vector<LedgerUpgrade>
-Upgrades::createUpgradesFor(LedgerHeader const& header) const
+Upgrades::createUpgradesFor(LedgerHeader const& lclHeader,
+                            AbstractLedgerTxn& ltx) const
 {
     auto result = std::vector<LedgerUpgrade>{};
-    if (!timeForUpgrade(header.scpValue.closeTime))
+    if (!timeForUpgrade(lclHeader.scpValue.closeTime))
```

Comment:
we also need to make sure we do not schedule invalid upgrades.
This is very important in the context of this PR.
So best is to change `createUpgradesFor` to not only check that the scheduled upgrade would do anything but also that it's valid

Reply:
added validity check